components/function-controller/pkg/controller/function/function_controller.go

```diff
@@ -611,56 +599,41 @@ func (r *ReconcileFunction) serveFunction(rnInfo *runtimeUtil.RuntimeInfo, found
 // - the conditions service, route and configuration should have status true and type ready.
 // Update the status of the function base on the defined function condition.
 // For a function get the status error either the creation or update of the knative service or build must have failed.
-func (r *ReconcileFunction) setFunctionCondition(fn *serverlessv1alpha1.Function) {
-	serviceReady := false
-	configurationsReady := false
-	routesReady := false
-
-	// Get the status of the Build
-	foundBuild := &buildv1alpha1.Build{}
-	if err := r.Get(context.TODO(), types.NamespacedName{Name: fn.Name, Namespace: fn.Namespace}, foundBuild); ignoreNotFound(err) != nil {
-		log.Error(err, "Error while trying to get the Knative Build for the Function Status", "namespace", fn.Namespace, "name", fn.Name)
-		return
-	}
-
-	// if build show error, set function status to error too
-	for _, condition := range foundBuild.Status.Conditions {
-		if condition.Type == duckv1alpha1.ConditionSucceeded && condition.Status == corev1.ConditionFalse {
-			err := r.updateFunctionStatus(fn, serverlessv1alpha1.FunctionConditionError)
-			if err != nil {
-				log.Error(err, "Error while trying to update the function Status", "namespace", fn.Namespace, "name", fn.Name)
+func (r *ReconcileFunction) setFunctionCondition(fn *serverlessv1alpha1.Function, tr *tektonv1alpha1.TaskRun,
```

Comment:
Notice how `setFunctionCondition()` now accepts the objects it needs to compute the Function status from as parameter, this way we can avoid calling `Get()` again for no reason. See previous comment.